pkg/k8s/resource/crdsync.go

```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package resource
+
+// CRDSync is a type used for promise.Promise, passing the error (if any) from the SyncCRDs() call
+// to the result of the promise Await() call.
```

Comment:
Isn't this what the `Reject(error)` function is for? i.e. What do we want to pass the error via the CRDType (with Resolve(err) sometimes vs Reject)?

Reply:
Good question! It would be weird to me is the CRDSync would be an empty type, so decided the pass the errror from `SyncCRDs` as the promise type. 

I'll test if this works with en empty type instead.